client/js/components/user/DpUserFormFields.vue

```diff
@@ -421,6 +451,47 @@ export default {
       if (this.isDepartmentSet === false) {
         this.setDefaultDepartment(userOrga)
       }
+    },
+
+    setOrga (payload) {
+      const payloadRel = payload.relationships
+      const payloadWithNewType = {
+        ...payload,
+        id: payload.id,
+        attributes: {
+          ...payload.attributes
+        },
+        // We have to hack it like this, because the types for relationships have to be in camelCase and not in PascalCase
+        relationships: {
+          allowedRoles: {
+            data: payloadRel.allowedRoles?.data[0].id
+              ? payloadRel.allowedRoles?.data.map(el => {
```

Comment:
You should omit the "?" between "allowedRoles" and ""data hence You already checked it the id is present in the line before